File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.py
def solution(triangle):
    answer = 0
    dp =[ [0]*i for i in range(1,len(triangle)+1)]
    dp[0] = triangle[0]
    
    for i in range(1,len(triangle)):
        for j in range(len(triangle[i])):
            if j == 0:
                dp[i][j] = triangle[i][j] + dp[i-1][j]
            elif i == j:
                dp[i][j] = triangle[i][j] + dp[i-1][j-1]
            else:
                dp[i][j] = max(dp[i-1][j], dp[i-1][j-1]) + triangle[i][j]
            
  
    
    return max(dp[-1])

File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.java
import java.util.*;
class Solution {
    public int solution(int[][] triangle) {
        int answer = 0;
        for(int i=triangle.length-1;i>-1;i--){
            
            for(int j=0; j<triangle[i].length-1;j++){
                int max = Math.max(triangle[i][j], triangle[i][j+1]);
                triangle[i-1][j] = triangle[i-1][j]+max;
            }
                
        }
        
        return triangle[0][0];
    }
}